src/www/ui_components/add-server-view.js

```diff
@@ -299,9 +306,11 @@ Polymer({
     input.toggleClass('input-invalid', input.invalid);
     if (input.invalid) {
```

Comment:
What's going on here? Is the key error replacing the message on the zero state screen? We usually use iron-pages to select one of many, which is scales a lot better.

Alternatively, have two variables: one for whether to show the alternative message (not affected by _inputInvalidChanged) and one for whether to show the error or zero state message (you already have it, it's `addServerFooter`).

Reply:
I understand this better now. Yeah, a variable for deciding on whether to show the error, and another to decide on which suggestion to show is cleaner.

Reply:
done, see below